lib/shoryuken/queue.rb

```diff
@@ -5,7 +5,11 @@ class Queue
     def initialize(client, name)
       self.name   = name
       self.client = client
-      self.url    = client.get_queue_url(queue_name: name).queue_url
+      begin
+        self.url = client.get_queue_url(queue_name: name).queue_url
+      rescue Aws::SQS::Errors::NonExistentQueue
+        raise StandardError.new "Non existing queue: #{name}. Please create it in advance in SQS."
```

Comment:
What about:

```ruby
rescue Aws::SQS::Errors::NonExistentQueue => e
  raise e.class, "#{e.message} (queue: #{name})", e.backtrace
```

This will keep the original error class and backtrace and will append the queue name to the error message.

A quick one-liner for testing in irb/pry:

```ruby
begin; raise ArgumentError, 'test'; rescue ArgumentError => e; raise e.class, "#{e.message} [modified]", e.backtrace; end
# ArgumentError: test [modified]
# from (pry):1:in `__pry__'
```